opencl-source-provider.cmake

```diff
@@ -0,0 +1,50 @@
+set(BDA_DIR opm/simulators/linalg/bda)
+set(KERNELS_DIR ${BDA_DIR}/opencl/kernels)
+
+if(DEBUG_KERNELS)
+  set(DEBUG_DIR ${KERNELS_DIR}/.debug)
+
+  execute_process(
+    COMMAND ${CMAKE_COMMAND} -E make_directory ${DEBUG_DIR}
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+  )
+endif()
+
+set(CL_SRC_FILE ${_leaf_dir_name}/clSources.cpp)
```

Comment:
This looks like an internal variable that might change without prior heads up. Would it be possible to use an official one like `${PROJECT_BINARY_DIR}`?

Reply:
Hopefully this will also fix the errors in jenkins:
```
CMake Error at opencl-source-provider.cmake:14 (file):
  file failed to open for writing (Permission denied):

    /clSources.cpp
Call Stack (most recent call first):
  CMakeLists.txt:291 (include)
  mpi/install/share/opm/cmake/Modules/OpmLibMain.cmake:120 (sources_hook)
  CMakeLists.txt:306 (include)
```

Reply:
reason might be deactivated SiblingSearch

Reply:
Ah, I forgot to change this. As soon as I get home I'll fix it.

Reply:
Fixed.